v-next/core/src/types/common.ts

```diff
@@ -57,11 +63,19 @@ const parameterTypeValidators: Record<
  *
  * This function uses a map of validators, where each validator is a function
  * that checks if a value is valid for a specific parameter type.
+ * If the parameter type is variadic, the value is considered valid if it is an
+ * array and all its elements are valid for the parameter type.
  */
 export function isParameterValueValid(
   type: ParameterType,
   value: unknown,
+  isVariadic: boolean = false,
 ): boolean {
   const validator = parameterTypeValidators[type];
```

Comment:
Should we use the same function used [here ](https://github.com/NomicFoundation/hardhat/blob/c9df7e6799b151643dce0bbcc94f3a79d0130544/v-next/hardhat/src/internal/cli/main.ts#L506C10-L506C29) to validate the different types?